packages/framework/build.xml

```diff
@@ -0,0 +1,1029 @@
+        <exec executable="${path.php.executable}" passthru="true" checkreturn="true">
+            <arg value="${path.bin-console}"/>
+            <arg value="shopsys:error-page:generate-all"/>
+        </exec>
+    </target>
+
+    <target name="generate-friendly-urls" description="Generates friendly urls for supported entities when missing.">
+        <exec executable="${path.php.executable}" passthru="true" checkreturn="true">
+            <arg value="${path.bin-console}"/>
+            <arg value="shopsys:generate:friendly-url"/>
+        </exec>
+    </target>
+
+    <target name="grunt" depends="gruntfile,clean-styles" description="Builds CSS from LESS via Grunt.">
+        <exec executable="${path.grunt.executable}" dir="${path.root}" passthru="true" checkreturn="true"/>
+    </target>
+
+    <target name="gruntfile" description="Generates Gruntfile.js by domain settings.">
+        <exec executable="${path.php.executable}" passthru="true" checkreturn="true">
+            <arg value="${path.bin-console}"/>
+            <arg value="shopsys:generate:gruntfile"/>
+        </exec>
+    </target>
+
+    <target name="list" description="Hidden target to make Phing list all targets when called without an argument." hidden="true">
+        <exec executable="${path.php.executable}" passthru="true" checkreturn="true">
+            <arg value="phing"/>
+            <arg value="-l"/>
+        </exec>
+    </target>
+
+    <target name="maintenance-off" description="Turns the maintenance page off.">
+        <delete file="${path.root}/MAINTENANCE"/>
+    </target>
+
+    <target name="maintenance-on" description="Turns the maintenance page on.">
+        <touch file="${path.root}/MAINTENANCE"/>
+    </target>
+
+    <target name="npm" description="Installs modules required for Grunt.">
+        <exec executable="${path.npm.executable}" dir="${path.root}" logoutput="true" passthru="true" checkreturn="true" >
+            <arg value="prune"/>
+        </exec>
+
+        <exec executable="${path.npm.executable}" dir="${path.root}" logoutput="true" passthru="true" checkreturn="true" >
+            <arg value="install"/>
+        </exec>
+    </target>
+
+    <target name="replace-domains-urls" description="Replaces domains urls in database by urls in configuration.">
+        <exec executable="${path.php.executable}" passthru="true" checkreturn="true">
+            <arg value="${path.bin-console}"/>
+            <arg value="shopsys:domains-urls:replace"/>
+        </exec>
+    </target>
+
+    <target name="load-plugin-demo-data" description="Loads data fixtures of all registered plugins." hidden="true">
+        <exec executable="${path.php.executable}" passthru="true" checkreturn="true">
+            <arg value="${path.bin-console}"/>
+            <arg value="shopsys:plugin-data-fixtures:load"/>
+        </exec>
+    </target>
+
+    <target name="warmup" description="Warms up cache.">
```

Comment:
`warmup` simply warms up the cache for current environment. `prod-warmup` is used before `error-pages-generate` because it needs to change the environment internally - see #816